diff --git a/mt7996/mmio.c b/mt7996/mmio.c
index 1d41be3b..aaaaaaaa 100644
--- a/mt7996/mmio.c
+++ b/mt7996/mmio.c
@@ -518,8 +518,11 @@
 		wed->wlan.id = MT7996_DEVICE_ID_2;
 		wed->wlan.tx_tbit[0] = ffs(MT_INT_TX_DONE_BAND2) - 1;
 	} else {
-		wed->wlan.hw_rro = mt7996_has_hwrro(dev);
+		if (dev->mt76.hwrro_mode == MT76_HWRRO_V3_1)
+			wed->wlan.hw_rro = MTK_WED_HWRRO_V3_1;
+		else if (dev->mt76.hwrro_mode == MT76_HWRRO_V3)
+			wed->wlan.hw_rro = MTK_WED_HWRRO_V3;
 		wed->wlan.wpdma_int = wed->wlan.phy_base + MT_INT_SOURCE_CSR;
 		wed->wlan.wpdma_mask = wed->wlan.phy_base + MT_INT_MASK_CSR;
 		wed->wlan.wpdma_tx = wed->wlan.phy_base + MT_TXQ_RING_BASE(0) +
 				     MT7996_TXQ_BAND0 * MT_RING_SIZE;
